### ISSUE TITLE
Allow to clear the proxy output window

### ISSUE THREAD

**Issue by waldekmastykarz** [microsoftgraph/msgraph-developer-proxy#19]
When debugging multiple page loads of a web app, it would be helpful to be able to clear the previous proxy output to more easily see what's current.

**Comment by waldekmastykarz** [comment]
Some terminals support this natively, so I wonder if we should implement this explicitly in the Proxy vs. leave it to the terminal to handle. Any opinion @gavinbarron, @garrytrinder, @sebastienlevert?

**Comment by sebastienlevert** [comment]
Can you share examples?

**Comment by waldekmastykarz** [comment]
In both iTerm2 and Warp on macOS I can use CMD+K to clear the terminal screen. Not sure if there's an equivalent in Terminal on Windows.

**Comment by sebastienlevert** [comment]
I think we should look at how Terminal (including both PowerShell and WSL) could allow this (I wasn't successful on PowerShell) as it would be a great opportunity.

That makes me think... Should we have a log of the current executed run? It leaves the terminal, but at least it offers an saved log of the guidance, endpoints, etc.

**Comment by waldekmastykarz** [comment]
> I think we should look at how Terminal (including both PowerShell and WSL) could allow this (I wasn't successful on PowerShell) as it would be a great opportunity.

We need to disambiguate between the terminal (iTerm2, Warp, Windows Terminal) and the shell (PowerShell, cmd, bash, zsh). If I'm not mistaken, clearing the output window is a terminal not a shell feature.

> That makes me think... Should we have a log of the current executed run? It leaves the terminal, but at least it offers an saved log of the guidance, endpoints, etc.

@gavinbarron and I talked about having support for different log sinks. Let's think if we'd prefer to have a full log of the output or a summary (eg. unique items, so if we returned $select guidance warning for /v1.0/me 5 times, we include it in the summary only once). Let's continue this discussion in a separate issue to more easily track the related work.

**Comment by waldekmastykarz** [comment]
I just found out that you can clear Windows Terminal window using CTRL+L. It clears only the visible portion of the window rather than clearing the whole buffer so let's see if that's sufficient

**Comment by garrytrinder** [comment]
> Some terminals support this natively, so I wonder if we should implement this explicitly in the Proxy vs. leave it to the terminal to handle. Any opinion @gavinbarron, @garrytrinder, @sebastienlevert?

If we can offload this to the terminal being used that would be ideal, if we do so we should capture the different keyboard shortcuts for the major terminals that folks could use in a How-to guide and add this to our Get started journey.

**Comment by waldekmastykarz** [comment]
> Just so I understand better. We're talking about clearing the output WHILE the proxy is running, right? Ctrl+L on windows seems to be only possible when nothing is currently running.

Ah, that's a bummer. In that case, we might still looking at a custom implementation.

**Comment by garrytrinder** [comment]
In v0.5.0 we will add the ability to start and stop recording using <kbd>r</kbd> and <kbd>s</kbd>.

Would expanding the list of keys to include <kbd>c</kbd> to clear the output be the next logical step?
